pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -71,11 +73,33 @@ func (k *KubeASCheck) Run() error {
 	if err != nil {
 		return err
 	}
+	// Only run if Leader Election is enabled.
+	if !config.Datadog.GetBool("leader_election") {
+		k.Warn("Leader Election not enabled. Not running Kubernetes API Server check or collecting Kubernetes Events.")
+		return nil
+	}
 
-	asclient, err := apiserver.GetAPIClient()
+	leaderEngine, err := leaderelection.GetLeaderEngine()
+	if err != nil {
+		k.Warn("Failed to instantiate the Leader Elector. Not running the Kubernetes API Server check or collecting Kubernetes Events.")
+		return err
+	}
 
+	err = leaderEngine.EnsureLeaderElectionRuns()
+	if err != nil {
+		k.Warn("Leader Election process failed to start")
+		return err
+	}
+
+	if !leaderEngine.IsLeader() {
+		log.Debugf("Leader is %s. %s will not run Kubernetes cluster related checks and collecting events", leaderEngine.CurrentLeaderName(), leaderEngine.HolderIdentity)
+		return nil
+	}
+	log.Tracef("%s is the Leader, running Kubernetes cluster related checks and collecting events", leaderEngine.CurrentLeaderName())
```

Comment:
this log & the log above carry the same info no?

Reply:
Not exactly the same, in the first one we notify the user we cannot run the kubernetes cluster check, events because not the leader.

In the second it's the opposite.

I'll change this to be more explicit.